graph-subscriptions-api/src/schema.rs

```diff
@@ -91,11 +93,81 @@ impl From<datasource::RequestTicket> for RequestTicketDto {
         Self {
             id,
             ticket_user: value.ticket_user,
-            ticket_signer: value.ticket_signer,
             ticket_name: value.ticket_name,
             total_query_count: value.total_query_count,
             queried_subgraphs_count: value.queried_subgraphs_count,
             last_query_timestamp: value.last_query_timestamp,
+            ticket_payload: value.ticket_payload.into(),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct TicketPayloadDto {
+    pub signer: String,
+    pub allowed_domains: Option<String>,
+    pub allowed_deployments: Option<String>,
+    pub allowed_subgraphs: Option<String>,
+}
+
+impl From<datasource::GatewaySubscriptionQueryResultTicketPayload> for TicketPayloadDto {
+    fn from(value: datasource::GatewaySubscriptionQueryResultTicketPayload) -> Self {
+        Self {
+            signer: value.signer,
+            allowed_domains: value.allowed_domains,
+            allowed_deployments: value.allowed_deployments,
+            allowed_subgraphs: value.allowed_subgraphs,
+        }
+    }
+}
+
+#[Object]
+impl TicketPayloadDto {
+    /// The ticket signer.
+    /// For most use-cases, this will match the `RequestTicket.ticket_user` value.
+    /// But if the signer is an authorized signer on the subscription, this value will be the signing wallet
+    pub async fn signer(&self) -> String {
+        self.signer.to_string()
+    }
+    /// A list of Subgraph Deployment Qm hashes the ticket is allowed to query
+    pub async fn allowed_deployments(&self) -> Vec<String> {
+        match &self.allowed_deployments {
+            Some(deployments) => deployments.split(",").map(|d| d.to_string()).collect(),
+            None => vec![],
+        }
+    }
+    /// A list of domains the ticket is allowed to query from
+    pub async fn allowed_domains(&self) -> Vec<String> {
+        match &self.allowed_domains {
+            Some(domains) => domains.split(",").map(|d| d.to_string()).collect(),
+            None => vec![],
+        }
+    }
+    /// A list of Subgraphs the ticket is allowed to query
+    pub async fn allowed_subgraphs<'ctx>(&self, ctx: &Context<'ctx>) -> Result<Vec<Subgraph>> {
+        match &self.allowed_subgraphs {
+            Some(subgraphs) => {
+                let schema_ctx = ctx
+                    .data_unchecked::<Arc<Mutex<GraphSubscriptionsSchemaCtx>>>()
+                    .lock()
+                    .await;
+                let subgraph_ids: Vec<SubgraphId> = subgraphs
+                    .split(",")
+                    .map(|s| SubgraphId::from_str(s).unwrap_or_default())
```

Comment:
This is defaulting malformed subgraph IDs to zeroed bytes. I think it would be less error-prone to filter out malformed values instead.
```suggestion
                    .filter_map(|s| SubgraphId::from_str(s).ok())
```

Reply:
Much cleaner. Thank you